System.register(["./vendor/svelte-CkBJVEsY.js","./vendor/carbon-components-svelte-DfRG9uuU.js","./vendor/carbon-icons-svelte-Boxs1M4A.js","./vendor/temml-BTciUaue.js"],function(e){"use strict";var t,n,r,s,a,o,i,l,c,p,_,d,u,m,g,f,$,h,b,x,v,y,C,S,w,R,M,L,N,E,T,A,I,U,P,k,B,V,G,D,O,z,j,F,H,q,K,W,X,Y,J,Q,Z,ee,te,ne,re,se,ae,oe,ie,le,ce,pe,_e,de,ue,me,ge,fe,$e,he,be,xe,ve,ye,Ce,Se,we,Re,Me,Le,Ne,Ee,Te,Ae;return{setters:[function(e){t=e.S,n=e.i,r=e.s,s=e.a3,a=e.d,o=e.a8,i=e.t,l=e.a,c=e.g,p=e.c,_=e.b,d=e.p,u=e.Q,m=e.B,g=e.e,f=e.K,$=e.M,h=e.L,b=e.z,x=e.I,v=e.T,y=e.U,C=e.D,S=e.q,w=e.F,R=e.G,M=e.N,L=e.y,N=e.k,E=e.u,T=e.l,A=e.m,I=e.h,U=e.o,P=e.a2,k=e.f,B=e.j,V=e.A,G=e.ac,D=e.n,O=e.W,z=e.ad,j=e.x,F=e.ae,H=e.R},function(e){q=e.S,K=e.a,W=e.b,X=e.H,Y=e.T,J=e.c,Q=e.d,Z=e.e,ee=e.f,te=e.g,ne=e.M,re=e.I,se=e.h,ae=e.i,oe=e.j,ie=e.C,le=e.k,ce=e.l,pe=e.G,_e=e.R,de=e.m,ue=e.B,me=e.n,ge=e.o,fe=e.A},function(e){$e=e.C,he=e.a,be=e.D,xe=e.A,ve=e.S,ye=e.I,Ce=e.b,Se=e.c,we=e.V,Re=e.d,Me=e.T,Le=e.G,Ne=e.J,Ee=e.P,Te=e.e,Ae=e.f},null],execute:function(){function Ie(e,t,n){const r=e.slice();return r[4]=t[n],r}function Ue(e,t,n){const r=e.slice();return r[7]=t[n],r}function Pe(e){let t,n;return t=new K({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text,expanded:!0,$$slots:{default:[Ve]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};1&n&&(r.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(r.class=e[2](e[4].page)),1&n&&(r.title=e[4].text),1&n&&(r.text=e[4].text),1025&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function ke(e){let t,n;return t=new q({props:{href:`${e[1](e[4].page)}#${e[4].slug}`,class:e[2](e[4].page),title:e[4].text,text:e[4].text}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};1&n&&(r.href=`${e[1](e[4].page)}#${e[4].slug}`),1&n&&(r.class=e[2](e[4].page)),1&n&&(r.title=e[4].text),1&n&&(r.text=e[4].text),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function Be(e){let t,n;return t=new W({props:{href:`${e[1](e[7].page)}#${e[7].slug}`,class:e[2](e[7].page),title:e[7].text,text:e[7].text}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};1&n&&(r.href=`${e[1](e[7].page)}#${e[7].slug}`),1&n&&(r.class=e[2](e[7].page)),1&n&&(r.title=e[7].text),1&n&&(r.text=e[7].text),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function Ve(e){let t,n,r=s(e[4].children),d=[];for(let t=0;t<r.length;t+=1)d[t]=Be(Ue(e,r,t));const u=e=>i(d[e],1,1,()=>{d[e]=null});return{c(){for(let e=0;e<d.length;e+=1)d[e].c();t=b()},m(e,r){for(let t=0;t<d.length;t+=1)d[t]&&d[t].m(e,r);_(e,t,r),n=!0},p(e,n){if(7&n){let a;for(r=s(e[4].children),a=0;a<r.length;a+=1){const s=Ue(e,r,a);d[a]?(d[a].p(s,n),l(d[a],1)):(d[a]=Be(s),d[a].c(),l(d[a],1),d[a].m(t.parentNode,t))}for(c(),a=r.length;a<d.length;a+=1)u(a);p()}},i(e){if(!n){for(let e=0;e<r.length;e+=1)l(d[e]);n=!0}},o(e){d=d.filter(Boolean);for(let e=0;e<d.length;e+=1)i(d[e]);n=!1},d(e){e&&a(t),o(d,e)}}}function Ge(e){let t,n,r,s;const o=[ke,Pe],d=[];function u(e,t){return 0==e[4].children.length?0:1}return t=u(e),n=d[t]=o[t](e),{c(){n.c(),r=g()},m(e,n){d[t].m(e,n),_(e,r,n),s=!0},p(e,s){let a=t;t=u(e),t===a?d[t].p(e,s):(c(),i(d[a],1,1,()=>{d[a]=null}),p(),n=d[t],n?n.p(e,s):(n=d[t]=o[t](e),n.c()),l(n,1),n.m(r.parentNode,r))},i(e){s||(l(n),s=!0)},o(e){i(n),s=!1},d(e){e&&a(r),d[t].d(e)}}}function De(e){let t,n,r=s(e[0]),m=[];for(let t=0;t<r.length;t+=1)m[t]=Ge(Ie(e,r,t));const g=e=>i(m[e],1,1,()=>{m[e]=null});return{c(){t=d("ul");for(let e=0;e<m.length;e+=1)m[e].c();u(t,"class","bx--side-nav__items pipen-report-sidenav svelte-12gpq3")},m(e,r){_(e,t,r);for(let e=0;e<m.length;e+=1)m[e]&&m[e].m(t,null);n=!0},p(e,[n]){if(7&n){let a;for(r=s(e[0]),a=0;a<r.length;a+=1){const s=Ie(e,r,a);m[a]?(m[a].p(s,n),l(m[a],1)):(m[a]=Ge(s),m[a].c(),l(m[a],1),m[a].m(t,null))}for(c(),a=r.length;a<m.length;a+=1)g(a);p()}},i(e){if(!n){for(let e=0;e<r.length;e+=1)l(m[e]);n=!0}},o(e){m=m.filter(Boolean);for(let e=0;e<m.length;e+=1)i(m[e]);n=!1},d(e){e&&a(t),o(m,e)}}}function Oe(e,t,n){let{toc:r}=t,{page:s}=t;return m(()=>{let e,t=new URL(window.location.href);e=t.hash.length>0?document.querySelector(`[href="${t.hash}"]`):document.querySelector(".toc-current-page"),e.scrollIntoView({behavior:"smooth"})}),e.$$set=e=>{"toc"in e&&n(0,r=e.toc),"page"in e&&n(3,s=e.page)},[r,e=>{if(s===e)return"";let t=new URL(window.location.href);t.hash="";const n=(t.searchParams.get("proc")||"_index").replace(/-\d+$/,"");return t.searchParams.set("proc",0===e?n:`${n}-${e}`),t.toString()},e=>e!=s?"toc-noncurrent-page":"toc-current-page",s]}class ze extends t{constructor(e){super(),n(this,e,Oe,De,r,{toc:0,page:3})}}function je(e,t,n){const r=e.slice();return r[7]=t[n],r}function Fe(e,t,n){const r=e.slice();return r[10]=t[n],r}function He(e){let t,n;return t=new Q({props:{isOpen:ct,transition:e[5],$$slots:{default:[rt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};8197&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function qe(e){let t;return{c(){t=R("Go to home:")},m(e,n){_(e,t,n)},d(e){e&&a(t)}}}function Ke(e){let t;return{c(){t=R(e[0])},m(e,n){_(e,t,n)},p(e,n){1&n&&w(t,e[0])},d(e){e&&a(t)}}}function We(e){let t;return{c(){t=R("Switch to process:")},m(e,n){_(e,t,n)},d(e){e&&a(t)}}}function Xe(e){let t,n;return t=new te({props:{title:e[7].name,href:`?proc=${e[7].name}`,$$slots:{default:[Je]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};4&n&&(r.title=e[7].name),4&n&&(r.href=`?proc=${e[7].name}`),8196&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function Ye(e){let t,n,r,d;t=new te({props:{class:"procgroup",$$slots:{default:[Qe]},$$scope:{ctx:e}}});let u=s(e[7].procs),m=[];for(let t=0;t<u.length;t+=1)m[t]=et(Fe(e,u,t));const x=e=>i(m[e],1,1,()=>{m[e]=null});return{c(){h(t.$$.fragment),n=b();for(let e=0;e<m.length;e+=1)m[e].c();r=g()},m(e,s){$(t,e,s),_(e,n,s);for(let t=0;t<m.length;t+=1)m[t]&&m[t].m(e,s);_(e,r,s),d=!0},p(e,n){const a={};if(8196&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a),4&n){let t;for(u=s(e[7].procs),t=0;t<u.length;t+=1){const s=Fe(e,u,t);m[t]?(m[t].p(s,n),l(m[t],1)):(m[t]=et(s),m[t].c(),l(m[t],1),m[t].m(r.parentNode,r))}for(c(),t=u.length;t<m.length;t+=1)x(t);p()}},i(e){if(!d){l(t.$$.fragment,e);for(let e=0;e<u.length;e+=1)l(m[e]);d=!0}},o(e){i(t.$$.fragment,e),m=m.filter(Boolean);for(let e=0;e<m.length;e+=1)i(m[e]);d=!1},d(e){e&&(a(n),a(r)),f(t,e),o(m,e)}}}function Je(e){let t,n,r=e[7].name+"";return{c(){t=R(r),n=b()},m(e,r){_(e,t,r),_(e,n,r)},p(e,n){4&n&&r!==(r=e[7].name+"")&&w(t,r)},d(e){e&&(a(t),a(n))}}}function Qe(e){let t,n=e[7].name+"";return{c(){t=R(n)},m(e,n){_(e,t,n)},p(e,r){4&r&&n!==(n=e[7].name+"")&&w(t,n)},d(e){e&&a(t)}}}function Ze(e){let t,n,r,s=e[10].name+"";return{c(){t=R("- "),n=R(s),r=b()},m(e,s){_(e,t,s),_(e,n,s),_(e,r,s)},p(e,t){4&t&&s!==(s=e[10].name+"")&&w(n,s)},d(e){e&&(a(t),a(n),a(r))}}}function et(e){let t,n;return t=new te({props:{title:e[10].name,class:"procgroup-proc",href:`?proc=${e[10].name}`,$$slots:{default:[Ze]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};4&n&&(r.title=e[10].name),4&n&&(r.href=`?proc=${e[10].name}`),8196&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function tt(e){let t,n,r,s;const o=[Ye,Xe],d=[];function u(e,t){return e[7].procs?0:1}return t=u(e),n=d[t]=o[t](e),{c(){n.c(),r=g()},m(e,n){d[t].m(e,n),_(e,r,n),s=!0},p(e,s){let a=t;t=u(e),t===a?d[t].p(e,s):(c(),i(d[a],1,1,()=>{d[a]=null}),p(),n=d[t],n?n.p(e,s):(n=d[t]=o[t](e),n.c()),l(n,1),n.m(r.parentNode,r))},i(e){s||(l(n),s=!0)},o(e){i(n),s=!1},d(e){e&&a(r),d[t].d(e)}}}function nt(e){let t,n,r,d,u,m,x,v;t=new ee({props:{$$slots:{default:[qe]},$$scope:{ctx:e}}}),r=new te({props:{href:"index.html",$$slots:{default:[Ke]},$$scope:{ctx:e}}}),u=new ee({props:{$$slots:{default:[We]},$$scope:{ctx:e}}});let y=s(e[2]),C=[];for(let t=0;t<y.length;t+=1)C[t]=tt(je(e,y,t));const S=e=>i(C[e],1,1,()=>{C[e]=null});return{c(){h(t.$$.fragment),n=b(),h(r.$$.fragment),d=b(),h(u.$$.fragment),m=b();for(let e=0;e<C.length;e+=1)C[e].c();x=g()},m(e,s){$(t,e,s),_(e,n,s),$(r,e,s),_(e,d,s),$(u,e,s),_(e,m,s);for(let t=0;t<C.length;t+=1)C[t]&&C[t].m(e,s);_(e,x,s),v=!0},p(e,n){const a={};8192&n&&(a.$$scope={dirty:n,ctx:e}),t.$set(a);const o={};8193&n&&(o.$$scope={dirty:n,ctx:e}),r.$set(o);const i={};if(8192&n&&(i.$$scope={dirty:n,ctx:e}),u.$set(i),4&n){let t;for(y=s(e[2]),t=0;t<y.length;t+=1){const r=je(e,y,t);C[t]?(C[t].p(r,n),l(C[t],1)):(C[t]=tt(r),C[t].c(),l(C[t],1),C[t].m(x.parentNode,x))}for(c(),t=y.length;t<C.length;t+=1)S(t);p()}},i(e){if(!v){l(t.$$.fragment,e),l(r.$$.fragment,e),l(u.$$.fragment,e);for(let e=0;e<y.length;e+=1)l(C[e]);v=!0}},o(e){i(t.$$.fragment,e),i(r.$$.fragment,e),i(u.$$.fragment,e),C=C.filter(Boolean);for(let e=0;e<C.length;e+=1)i(C[e]);v=!1},d(e){e&&(a(n),a(d),a(m),a(x)),f(t,e),f(r,e),f(u,e),o(C,e)}}}function rt(e){let t,n;return t=new Z({props:{$$slots:{default:[nt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};8197&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function st(e){let t,n,r,s;n=new Y({props:{render:"select",theme:"g10",persist:!0,persistKey:"__pipen-report-theme",select:{themes:["g10","white","g80","g90","g100"],labelText:"Theme",inline:!0}}});let o=e[2].length>0&&He(e);return{c(){t=d("div"),h(n.$$.fragment),r=b(),o&&o.c(),u(t,"class","header-util"),S(t,"bx--header__global",!0)},m(e,a){_(e,t,a),$(n,t,null),C(t,r),o&&o.m(t,null),s=!0},p(e,n){e[2].length>0?o?(o.p(e,n),4&n&&l(o,1)):(o=He(e),o.c(),l(o,1),o.m(t,null)):o&&(c(),i(o,1,1,()=>{o=null}),p())},i(e){s||(l(n.$$.fragment,e),l(o),s=!0)},o(e){i(n.$$.fragment,e),i(o),s=!1},d(e){e&&a(t),f(n),o&&o.d()}}}function at(e){let t;return{c(){t=R(e[4])},m(e,n){_(e,t,n)},p(e,n){16&n&&w(t,e[4])},d(e){e&&a(t)}}}function ot(e){let t,n,r,s,o=null!==e[4]&&at(e);return{c(){t=d("h1"),n=R(e[3]),r=b(),o&&o.c(),s=g()},m(e,a){_(e,t,a),C(t,n),_(e,r,a),o&&o.m(e,a),_(e,s,a)},p(e,t){8&t&&w(n,e[3]),null!==e[4]?o?o.p(e,t):(o=at(e),o.c(),o.m(s.parentNode,s)):o&&(o.d(1),o=null)},d(e){e&&(a(t),a(r),a(s)),o&&o.d(e)}}}function it(e){let t,n;return t=new J({props:{class:"header-tile",slot:"platform",$$slots:{default:[ot]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};8216&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function lt(e){let t,n,r;function s(t){e[6](t)}let a={$$slots:{platform:[it],default:[st]},$$scope:{ctx:e}};return void 0!==e[1]&&(a.isSideNavOpen=e[1]),t=new X({props:a}),x.push(()=>v(t,"isSideNavOpen",s)),{c(){h(t.$$.fragment)},m(e,n){$(t,e,n),r=!0},p(e,[r]){const s={};8221&r&&(s.$$scope={dirty:r,ctx:e}),!n&&2&r&&(n=!0,s.isSideNavOpen=e[1],y(()=>n=!1)),t.$set(s)},i(e){r||(l(t.$$.fragment,e),r=!0)},o(e){i(t.$$.fragment,e),r=!1},d(e){f(t,e)}}}let ct=!1;function pt(e,t,n){let{entries:r}=t,{logo:s}=t,{logotext:a=null}=t,{pipeline_name:o}=t,{isSideNavOpen:i}=t;return o=o||s,e.$$set=e=>{"entries"in e&&n(2,r=e.entries),"logo"in e&&n(3,s=e.logo),"logotext"in e&&n(4,a=e.logotext),"pipeline_name"in e&&n(0,o=e.pipeline_name),"isSideNavOpen"in e&&n(1,i=e.isSideNavOpen)},[o,i,r,s,a,{duration:200},function(e){i=e,n(1,i)}]}class _t extends t{constructor(e){super(),n(this,e,pt,lt,r,{entries:2,logo:3,logotext:4,pipeline_name:0,isSideNavOpen:1})}}function dt(e){let t,n,r;function s(t){e[3](t)}let a={modalHeading:"Running Information",passiveModal:!0,hasScrollingContent:!0,class:"runinfo",$$slots:{default:[ht]},$$scope:{ctx:e}};return void 0!==e[2]&&(a.open=e[2]),t=new ne({props:a}),x.push(()=>v(t,"open",s)),{c(){h(t.$$.fragment)},m(e,n){$(t,e,n),r=!0},p(e,r){const s={};34&r&&(s.$$scope={dirty:r,ctx:e}),!n&&4&r&&(n=!0,s.open=e[2],y(()=>n=!1)),t.$set(s)},i(e){r||(l(t.$$.fragment,e),r=!0)},o(e){i(t.$$.fragment,e),r=!1},d(e){f(t,e)}}}function ut(e){let t,n,r,s,o,c;return t=new ae({props:{label:"Session Information"}}),r=new ae({props:{label:"Time Spent"}}),o=new ae({props:{label:"Device Information"}}),{c(){h(t.$$.fragment),n=b(),h(r.$$.fragment),s=b(),h(o.$$.fragment)},m(e,a){$(t,e,a),_(e,n,a),$(r,e,a),_(e,s,a),$(o,e,a),c=!0},p:M,i(e){c||(l(t.$$.fragment,e),l(r.$$.fragment,e),l(o.$$.fragment,e),c=!0)},o(e){i(t.$$.fragment,e),i(r.$$.fragment,e),i(o.$$.fragment,e),c=!1},d(e){e&&(a(n),a(s)),f(t,e),f(r,e),f(o,e)}}}function mt(e){let t,n;return t=new ie({props:{type:"multi",code:e[1].session}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};2&n&&(r.code=e[1].session),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function gt(e){let t,n;return t=new ie({props:{type:"multi",code:e[1].time}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};2&n&&(r.code=e[1].time),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function ft(e){let t,n;return t=new ie({props:{type:"multi",code:e[1].device}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};2&n&&(r.code=e[1].device),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function $t(e){let t,n,r,s,o,c;return t=new oe({props:{$$slots:{default:[mt]},$$scope:{ctx:e}}}),r=new oe({props:{$$slots:{default:[gt]},$$scope:{ctx:e}}}),o=new oe({props:{$$slots:{default:[ft]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment),n=b(),h(r.$$.fragment),s=b(),h(o.$$.fragment)},m(e,a){$(t,e,a),_(e,n,a),$(r,e,a),_(e,s,a),$(o,e,a),c=!0},p(e,n){const s={};34&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const a={};34&n&&(a.$$scope={dirty:n,ctx:e}),r.$set(a);const i={};34&n&&(i.$$scope={dirty:n,ctx:e}),o.$set(i)},i(e){c||(l(t.$$.fragment,e),l(r.$$.fragment,e),l(o.$$.fragment,e),c=!0)},o(e){i(t.$$.fragment,e),i(r.$$.fragment,e),i(o.$$.fragment,e),c=!1},d(e){e&&(a(n),a(s)),f(t,e),f(r,e),f(o,e)}}}function ht(e){let t,n,r,s;return t=new re({props:{kind:"warning",lowContrast:!0,hideCloseButton:!0,title:"Showing the first job only. Check the workdir for information of other jobs if any."}}),r=new se({props:{$$slots:{content:[$t],default:[ut]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment),n=b(),h(r.$$.fragment)},m(e,a){$(t,e,a),_(e,n,a),$(r,e,a),s=!0},p(e,t){const n={};34&t&&(n.$$scope={dirty:t,ctx:e}),r.$set(n)},i(e){s||(l(t.$$.fragment,e),l(r.$$.fragment,e),s=!0)},o(e){i(t.$$.fragment,e),i(r.$$.fragment,e),s=!1},d(e){e&&a(n),f(t,e),f(r,e)}}}function bt(e){let t,n,r;return{c(){t=d("a"),t.textContent="> Running Information",u(t,"href","javascript:void(0);")},m(s,a){_(s,t,a),n||(r=L(t,"click",e[4]),n=!0)},p:M,d(e){e&&a(t),n=!1,r()}}}function xt(e){let t,n,r,s,o,m,g,f,$=e[1]&&dt(e),h=e[1]&&bt(e);return{c(){$&&$.c(),t=b(),n=d("div"),r=d("div"),h&&h.c(),s=b(),o=d("div"),m=R("Powered by "),g=d("span"),u(r,"class","runinfo"),u(g,"class","pipen-report-footer-versions"),u(o,"class","versions"),u(n,"class","pipen-report-footer bx--grid")},m(a,i){$&&$.m(a,i),_(a,t,i),_(a,n,i),C(n,r),h&&h.m(r,null),C(n,s),C(n,o),C(o,m),C(o,g),g.innerHTML=e[0],f=!0},p(e,[n]){e[1]?$?($.p(e,n),2&n&&l($,1)):($=dt(e),$.c(),l($,1),$.m(t.parentNode,t)):$&&(c(),i($,1,1,()=>{$=null}),p()),e[1]?h?h.p(e,n):(h=bt(e),h.c(),h.m(r,null)):h&&(h.d(1),h=null),(!f||1&n)&&(g.innerHTML=e[0])},i(e){f||(l($),f=!0)},o(e){i($),f=!1},d(e){e&&(a(t),a(n)),$&&$.d(e),h&&h.d()}}}function vt(e,t,n){let{versions:r}=t,{runinfo:s}=t,a=!1;return e.$$set=e=>{"versions"in e&&n(0,r=e.versions),"runinfo"in e&&n(1,s=e.runinfo)},[r,s,a,function(e){a=e,n(2,a)},()=>{n(2,a=!0)}]}class yt extends t{constructor(e){super(),n(this,e,vt,xt,r,{versions:0,runinfo:1})}}function Ct(e){let t,n,r;function s(t){e[12](t)}let a={$$slots:{default:[St]},$$scope:{ctx:e}};return void 0!==e[0]&&(a.isOpen=e[0]),t=new ce({props:a}),x.push(()=>v(t,"isOpen",s)),{c(){h(t.$$.fragment)},m(e,n){$(t,e,n),r=!0},p(e,r){const s={};8576&r&&(s.$$scope={dirty:r,ctx:e}),!n&&1&r&&(n=!0,s.isOpen=e[0],y(()=>n=!1)),t.$set(s)},i(e){r||(l(t.$$.fragment,e),r=!0)},o(e){i(t.$$.fragment,e),r=!1},d(e){f(t,e)}}}function St(e){let t,n;return t=new ze({props:{toc:e[7],page:e[8]}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};128&n&&(r.toc=e[7]),256&n&&(r.page=e[8]),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function wt(e){let t;const n=e[10].default,r=N(n,e,e[13],null);return{c(){r&&r.c()},m(e,n){r&&r.m(e,n),t=!0},p(e,s){r&&r.p&&(!t||8192&s)&&E(r,n,e,e[13],t?A(n,e[13],s,null):T(e[13]),null)},i(e){t||(l(r,e),t=!0)},o(e){i(r,e),t=!1},d(e){r&&r.d(e)}}}function Rt(e){let t,n;return t=new de({props:{$$slots:{default:[wt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};8192&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function Mt(e){let t,n;return t=new _e({props:{$$slots:{default:[Rt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};8192&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function Lt(e){let t,n,r,s;return t=new pe({props:{$$slots:{default:[Mt]},$$scope:{ctx:e}}}),r=new yt({props:{versions:e[3],runinfo:e[9]}}),{c(){h(t.$$.fragment),n=b(),h(r.$$.fragment)},m(e,a){$(t,e,a),_(e,n,a),$(r,e,a),s=!0},p(e,n){const s={};8192&n&&(s.$$scope={dirty:n,ctx:e}),t.$set(s);const a={};8&n&&(a.versions=e[3]),512&n&&(a.runinfo=e[9]),r.$set(a)},i(e){s||(l(t.$$.fragment,e),l(r.$$.fragment,e),s=!0)},o(e){i(t.$$.fragment,e),i(r.$$.fragment,e),s=!1},d(e){e&&a(n),f(t,e),f(r,e)}}}function Nt(e){let t,n,r,s,o,d;function u(t){e[11](t)}let m={logo:e[1],logotext:e[2],entries:e[4],pipeline_name:e[5]};void 0!==e[0]&&(m.isSideNavOpen=e[0]),t=new _t({props:m}),x.push(()=>v(t,"isSideNavOpen",u));let g=e[6]&&e[7].length>0&&Ct(e);return o=new le({props:{$$slots:{default:[Lt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment),r=b(),g&&g.c(),s=b(),h(o.$$.fragment)},m(e,n){$(t,e,n),_(e,r,n),g&&g.m(e,n),_(e,s,n),$(o,e,n),d=!0},p(e,[r]){const a={};2&r&&(a.logo=e[1]),4&r&&(a.logotext=e[2]),16&r&&(a.entries=e[4]),32&r&&(a.pipeline_name=e[5]),!n&&1&r&&(n=!0,a.isSideNavOpen=e[0],y(()=>n=!1)),t.$set(a),e[6]&&e[7].length>0?g?(g.p(e,r),192&r&&l(g,1)):(g=Ct(e),g.c(),l(g,1),g.m(s.parentNode,s)):g&&(c(),i(g,1,1,()=>{g=null}),p());const _={};8712&r&&(_.$$scope={dirty:r,ctx:e}),o.$set(_)},i(e){d||(l(t.$$.fragment,e),l(g),l(o.$$.fragment,e),d=!0)},o(e){i(t.$$.fragment,e),i(g),i(o.$$.fragment,e),d=!1},d(e){e&&(a(r),a(s)),f(t,e),g&&g.d(e),f(o,e)}}}function Et(e,t,n){let r,{$$slots:s={},$$scope:a}=t,{logo:o}=t,{logotext:i=null}=t,{versions:l}=t,{entries:c}=t,{pipeline_name:p}=t,{report_toc:_}=t,{toc:d}=t,{page:u}=t,{isSideNavOpen:g}=t,{runinfo:f}=t;return m(()=>{const e=new URL(window.location.href);let t;t=e.hash.length>0?document.querySelector(`li:has(>[href="${e.hash}"])`):document.querySelector("li:has(>.toc-current-page)"),t&&(t.classList.add("toc-active"),t.scrollIntoView());const n=document.querySelectorAll(".pipen-report-toc-anchor");r=new IntersectionObserver(e=>{e.forEach(e=>{const t=e.target.getAttribute("id"),n=document.querySelector(`li:has(>[href="#${t}"])`);n&&(e.isIntersecting?n.classList.add("toc-active"):n.classList.remove("toc-active"))})},{root:null,rootMargin:"0px",threshold:.5}),n.forEach(e=>{r.observe(e)})}),e.$$set=e=>{"logo"in e&&n(1,o=e.logo),"logotext"in e&&n(2,i=e.logotext),"versions"in e&&n(3,l=e.versions),"entries"in e&&n(4,c=e.entries),"pipeline_name"in e&&n(5,p=e.pipeline_name),"report_toc"in e&&n(6,_=e.report_toc),"toc"in e&&n(7,d=e.toc),"page"in e&&n(8,u=e.page),"isSideNavOpen"in e&&n(0,g=e.isSideNavOpen),"runinfo"in e&&n(9,f=e.runinfo),"$$scope"in e&&n(13,a=e.$$scope)},[g,o,i,l,c,p,_,d,u,f,s,function(e){g=e,n(0,g)},function(e){g=e,n(0,g)},a]}class Tt extends t{constructor(e){super(),n(this,e,Et,Nt,r,{logo:1,logotext:2,versions:3,entries:4,pipeline_name:5,report_toc:6,toc:7,page:8,isSideNavOpen:0,runinfo:9})}}function At(e){let t;return{c(){t=R(e[2])},m(e,n){_(e,t,n)},p:M,d(e){e&&a(t)}}}function It(e){let t,n,r,s;return n=new ue({props:{icon:e[1],kind:"tertiary",href:e[3](),$$slots:{default:[At]},$$scope:{ctx:e}}}),{c(){t=d("div"),h(n.$$.fragment),u(t,"class",r=`pagenav ${e[0]}`)},m(e,r){_(e,t,r),$(n,t,null),s=!0},p(e,[a]){const o={};16&a&&(o.$$scope={dirty:a,ctx:e}),n.$set(o),(!s||1&a&&r!==(r=`pagenav ${e[0]}`))&&u(t,"class",r)},i(e){s||(l(n.$$.fragment,e),s=!0)},o(e){i(n.$$.fragment,e),s=!1},d(e){e&&a(t),f(n)}}}function Ut(e,t,n){let{dir:r}=t,s="up"===r?$e:he,a="up"===r?"Go to previous page":"Go to next page";return e.$$set=e=>{"dir"in e&&n(0,r=e.dir)},[r,s,a,()=>{let e=new URL(window.location.href);e.hash="";const t=e.searchParams.get("proc")||"_index",n=parseInt((t.match(/-(\d+)$/)||["-0","0"])[1]),s="up"===r?n-1:n+1,a=t.replace(/-\d+$/,""),o=0===s?a:`${a}-${s}`;return e.searchParams.set("proc",o),e.toString()}]}class Pt extends t{constructor(e){super(),n(this,e,Ut,It,r,{dir:0})}}function kt(e){let t,n;const r=[{tooltipText:e[0]},{icon:e[1]},{direction:"bottom"},{align:"start"},e[3]];let s={};for(let e=0;e<r.length;e+=1)s=I(s,r[e]);return t=new me({props:s}),t.$on("click",e[2]),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,[n]){const s=11&n?U(r,[1&n&&{tooltipText:e[0]},2&n&&{icon:e[1]},r[2],r[3],8&n&&P(e[3])]):{};t.$set(s)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function Bt(e,t,n){const r=["src","tip","icon"];let s,a=k(t,r),{src:o=""}=t,{tip:i=""}=t,{icon:l="auto"}=t,c=o.split(".").at(-1).toLocaleLowerCase();if("string"==typeof l)switch("auto"===l&&(l=c),l){case"pdf":case"DocumentPdf":s=Ae;break;case"html":case"htm":case"ChartLineData":s=Te;break;case"png":case"Png":s=Ee;break;case"jpg":case"jpeg":case"Jpg":s=Ne;break;case"gif":case"Gif":s=Le;break;case"tif":case"tiff":case"Tif":s=Me;break;case"eps":case"DatabaseElastic":s=Re;break;case"mp4":case"avi":case"mov":case"mkv":case"webm":case"VideoPlayer":s=we;break;case"Image":s=Se;break;case"Data":case"DataVolume":s=Ce;break;case"Code":case"IbmWatsonxCodeAssistantForZRefactor":s=ye;break;case"svg":case"Svg":s=ve;break;case"zip":case"tar":case"gz":case"bz2":case"7z":case"rar":case"xz":case"Archive":s=xe;break;default:s=be}const p={...a,class:[a.class,"pipen-report-image-download"].filter(Boolean).join(" ")};return e.$$set=e=>{t=I(I({},t),B(e)),n(7,a=k(t,r)),"src"in e&&n(5,o=e.src),"tip"in e&&n(0,i=e.tip),"icon"in e&&n(4,l=e.icon)},[i,s,()=>{const e=document.createElement("a");e.href=o,e.download=o.split("/").pop(),e.click()},p,l,o]}class Vt extends t{constructor(e){super(),n(this,e,Bt,kt,r,{src:5,tip:0,icon:4})}}function Gt(e){let t,n,r;return{c(){t=R(e[1]),n=R("x"),r=R(e[2])},m(e,s){_(e,t,s),_(e,n,s),_(e,r,s)},p(e,n){2&n&&w(t,e[1]),4&n&&w(r,e[2])},d(e){e&&(a(t),a(n),a(r))}}}function Dt(e){let t,n;return t=new ge({props:{tooltipText:e[0],class:"pipen-report-image-dimension",align:"start",direction:"bottom",$$slots:{default:[Gt]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,[n]){const r={};1&n&&(r.tooltipText=e[0]),38&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function Ot(e,t,n){let{src:r=""}=t,{tip:s=""}=t,a=0,o=0,i=new Image;return i.src=r,i.onload=()=>{n(1,a=i.width),n(2,o=i.height)},e.$$set=e=>{"src"in e&&n(3,r=e.src),"tip"in e&&n(0,s=e.tip)},[s,a,o,r]}class zt extends t{constructor(e){super(),n(this,e,Ot,Dt,r,{src:3,tip:0})}}const{Boolean:jt}=G;function Ft(e,t,n){const r=e.slice();return r[6]=t[n].src,r[7]=t[n].tip,r[8]=t[n].icon,r[10]=n,r}function Ht(e){let t,n,r,u,m=s(e[0]),g=[];for(let t=0;t<m.length;t+=1)g[t]=Wt(Ft(e,m,t));const f=e=>i(g[e],1,1,()=>{g[e]=null});let $=[e[2],{role:"button"},{tabindex:"0"}],h={};for(let e=0;e<$.length;e+=1)h=I(h,$[e]);return{c(){t=d("div");for(let e=0;e<g.length;e+=1)g[e].c();D(t,h),S(t,"svelte-14lcnln",!0)},m(s,a){_(s,t,a);for(let e=0;e<g.length;e+=1)g[e]&&g[e].m(t,null);n=!0,r||(u=L(t,"mouseenter",e[4]),r=!0)},p(e,n){if(3&n){let r;for(m=s(e[0]),r=0;r<m.length;r+=1){const s=Ft(e,m,r);g[r]?(g[r].p(s,n),l(g[r],1)):(g[r]=Wt(s),g[r].c(),l(g[r],1),g[r].m(t,null))}for(c(),r=m.length;r<g.length;r+=1)f(r);p()}S(t,"svelte-14lcnln",!0)},i(e){if(!n){for(let e=0;e<m.length;e+=1)l(g[e]);n=!0}},o(e){g=g.filter(jt);for(let e=0;e<g.length;e+=1)i(g[e]);n=!1},d(e){e&&a(t),o(g,e),r=!1,u()}}}function qt(e){let t,n;return t=new Vt({props:{src:e[6],tip:e[7],icon:e[8]}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};1&n&&(r.src=e[6]),1&n&&(r.tip=e[7]),1&n&&(r.icon=e[8]),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function Kt(e){let t,n;return t=new zt({props:{src:e[1],tip:e[7]}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};2&n&&(r.src=e[1]),1&n&&(r.tip=e[7]),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function Wt(e){let t,n,r,s;const o=[Kt,qt],d=[];function u(e,t){return"x"===e[6]?0:1}return t=u(e),n=d[t]=o[t](e),{c(){n.c(),r=g()},m(e,n){d[t].m(e,n),_(e,r,n),s=!0},p(e,s){let a=t;t=u(e),t===a?d[t].p(e,s):(c(),i(d[a],1,1,()=>{d[a]=null}),p(),n=d[t],n?n.p(e,s):(n=d[t]=o[t](e),n.c()),l(n,1),n.m(r.parentNode,r))},i(e){s||(l(n),s=!0)},o(e){i(n),s=!1},d(e){e&&a(r),d[t].d(e)}}}function Xt(e){let t,n,r=e[0].length>0&&Ht(e);return{c(){r&&r.c(),t=g()},m(e,s){r&&r.m(e,s),_(e,t,s),n=!0},p(e,[n]){e[0].length>0?r?(r.p(e,n),1&n&&l(r,1)):(r=Ht(e),r.c(),l(r,1),r.m(t.parentNode,t)):r&&(c(),i(r,1,1,()=>{r=null}),p())},i(e){n||(l(r),n=!0)},o(e){i(r),n=!1},d(e){e&&a(t),r&&r.d(e)}}}function Yt(e,t,n){const r=["data","dsrc"];let s=k(t,r),{data:a=[]}=t,{dsrc:o=""}=t;Array.isArray(a)||(a=[a]),a=[{src:"x",tip:"widthxheight"},...a],a=a.map(e=>"string"==typeof e?{src:e,tip:`Download the ${e.split(".").at(-1)} format.`,icon:"auto"}:(e.tip=e.tip||`Download the ${e.src.split(".").at(-1)} format.`,e.icon=e.icon||"auto",e));const i={...s,class:[s.class,"pipen-report-image-download-list"].filter(Boolean).join(" ")},l=V();return e.$$set=e=>{t=I(I({},t),B(e)),n(5,s=k(t,r)),"data"in e&&n(0,a=e.data),"dsrc"in e&&n(1,o=e.dsrc)},[a,o,i,l,()=>l("mouseenter")]}class Jt extends t{constructor(e){super(),n(this,e,Yt,Xt,r,{data:0,dsrc:1})}}const Qt=e=>({}),Zt=e=>({}),en=e=>({}),tn=e=>({}),nn=e=>({}),rn=e=>({}),sn=e=>({}),an=e=>({});function on(e){let t,n;return t=new fe({props:{ratio:e[6],$$slots:{default:[un]},$$scope:{ctx:e}}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};64&n&&(r.ratio=e[6]),33556159&n&&(r.$$scope={dirty:n,ctx:e}),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function ln(e){let t,n,r,s,o=e[2]&&mn(e),d=e[0]&&gn(e),u=e[1]&&$n(e);return{c(){o&&o.c(),t=b(),d&&d.c(),n=b(),u&&u.c(),r=g()},m(e,a){o&&o.m(e,a),_(e,t,a),d&&d.m(e,a),_(e,n,a),u&&u.m(e,a),_(e,r,a),s=!0},p(e,s){e[2]?o?(o.p(e,s),4&s&&l(o,1)):(o=mn(e),o.c(),l(o,1),o.m(t.parentNode,t)):o&&(c(),i(o,1,1,()=>{o=null}),p()),e[0]?d?(d.p(e,s),1&s&&l(d,1)):(d=gn(e),d.c(),l(d,1),d.m(n.parentNode,n)):d&&(c(),i(d,1,1,()=>{d=null}),p()),e[1]?u?(u.p(e,s),2&s&&l(u,1)):(u=$n(e),u.c(),l(u,1),u.m(r.parentNode,r)):u&&(c(),i(u,1,1,()=>{u=null}),p())},i(e){s||(l(o),l(d),l(u),s=!0)},o(e){i(o),i(d),i(u),s=!1},d(e){e&&(a(t),a(n),a(r)),o&&o.d(e),d&&d.d(e),u&&u.d(e)}}}function cn(e){let t;const n=e[15].loading,r=N(n,e,e[25],tn);return{c(){r&&r.c()},m(e,n){r&&r.m(e,n),t=!0},p(e,s){r&&r.p&&(!t||33554432&s)&&E(r,n,e,e[25],t?A(n,e[25],s,en):T(e[25]),tn)},i(e){t||(l(r,e),t=!0)},o(e){i(r,e),t=!1},d(e){r&&r.d(e)}}}function pn(e){let t,n,r,s,o,u,m,f,$=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],h={};for(let e=0;e<$.length;e+=1)h=I(h,$[e]);let x=e[9]&&_n(e);return{c(){t=d("img"),s=b(),x&&x.c(),o=g(),D(t,h)},m(n,r){_(n,t,r),_(n,s,r),x&&x.m(n,r),_(n,o,r),u=!0,m||(f=[L(t,"click",e[11]),L(t,"mouseenter",e[19]),L(t,"mouseleave",e[20])],m=!0)},p(r,s){D(t,h=U($,[(e=r)[12],(!u||8&s&&!F(t.src,n=e[3]))&&{src:n},(!u||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?x?(x.p(e,s),512&s&&l(x,1)):(x=_n(e),x.c(),l(x,1),x.m(o.parentNode,o)):x&&(c(),i(x,1,1,()=>{x=null}),p())},i(n){u||(n&&j(()=>{u&&(r||(r=O(t,z,{duration:e[7]?xn:0},!0)),r.run(1))}),l(x),u=!0)},o(n){n&&(r||(r=O(t,z,{duration:e[7]?xn:0},!1)),r.run(0)),i(x),u=!1},d(e){e&&(a(t),a(s),a(o)),e&&r&&r.end(),x&&x.d(e),m=!1,H(f)}}}function _n(e){let t,n;return t=new Jt({props:{data:e[5]}}),t.$on("mouseenter",e[21]),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};32&n&&(r.data=e[5]),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function dn(e){let t;const n=e[15].error,r=N(n,e,e[25],Zt);return{c(){r&&r.c()},m(e,n){r&&r.m(e,n),t=!0},p(e,s){r&&r.p&&(!t||33554432&s)&&E(r,n,e,e[25],t?A(n,e[25],s,Qt):T(e[25]),Zt)},i(e){t||(l(r,e),t=!0)},o(e){i(r,e),t=!1},d(e){r&&r.d(e)}}}function un(e){let t,n,r,s,o=e[2]&&cn(e),d=e[0]&&pn(e),u=e[1]&&dn(e);return{c(){o&&o.c(),t=b(),d&&d.c(),n=b(),u&&u.c(),r=g()},m(e,a){o&&o.m(e,a),_(e,t,a),d&&d.m(e,a),_(e,n,a),u&&u.m(e,a),_(e,r,a),s=!0},p(e,s){e[2]?o?(o.p(e,s),4&s&&l(o,1)):(o=cn(e),o.c(),l(o,1),o.m(t.parentNode,t)):o&&(c(),i(o,1,1,()=>{o=null}),p()),e[0]?d?(d.p(e,s),1&s&&l(d,1)):(d=pn(e),d.c(),l(d,1),d.m(n.parentNode,n)):d&&(c(),i(d,1,1,()=>{d=null}),p()),e[1]?u?(u.p(e,s),2&s&&l(u,1)):(u=dn(e),u.c(),l(u,1),u.m(r.parentNode,r)):u&&(c(),i(u,1,1,()=>{u=null}),p())},i(e){s||(l(o),l(d),l(u),s=!0)},o(e){i(o),i(d),i(u),s=!1},d(e){e&&(a(t),a(n),a(r)),o&&o.d(e),d&&d.d(e),u&&u.d(e)}}}function mn(e){let t;const n=e[15].loading,r=N(n,e,e[25],an);return{c(){r&&r.c()},m(e,n){r&&r.m(e,n),t=!0},p(e,s){r&&r.p&&(!t||33554432&s)&&E(r,n,e,e[25],t?A(n,e[25],s,sn):T(e[25]),an)},i(e){t||(l(r,e),t=!0)},o(e){i(r,e),t=!1},d(e){r&&r.d(e)}}}function gn(e){let t,n,r,s,o,u,m,f,$=[e[12],{src:n=e[3]},{alt:e[4]},{title:"Click to zoom in ..."}],h={};for(let e=0;e<$.length;e+=1)h=I(h,$[e]);let x=e[9]&&fn(e);return{c(){t=d("img"),s=b(),x&&x.c(),o=g(),D(t,h)},m(n,r){_(n,t,r),_(n,s,r),x&&x.m(n,r),_(n,o,r),u=!0,m||(f=[L(t,"click",e[11]),L(t,"mouseenter",e[16]),L(t,"mouseleave",e[17])],m=!0)},p(r,s){D(t,h=U($,[(e=r)[12],(!u||8&s&&!F(t.src,n=e[3]))&&{src:n},(!u||16&s)&&{alt:e[4]},{title:"Click to zoom in ..."}])),e[9]?x?(x.p(e,s),512&s&&l(x,1)):(x=fn(e),x.c(),l(x,1),x.m(o.parentNode,o)):x&&(c(),i(x,1,1,()=>{x=null}),p())},i(n){u||(n&&j(()=>{u&&(r||(r=O(t,z,{duration:e[7]?xn:0},!0)),r.run(1))}),l(x),u=!0)},o(n){n&&(r||(r=O(t,z,{duration:e[7]?xn:0},!1)),r.run(0)),i(x),u=!1},d(e){e&&(a(t),a(s),a(o)),e&&r&&r.end(),x&&x.d(e),m=!1,H(f)}}}function fn(e){let t,n;return t=new Jt({props:{dsrc:e[3],data:e[5]}}),t.$on("mouseenter",e[18]),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},p(e,n){const r={};8&n&&(r.dsrc=e[3]),32&n&&(r.data=e[5]),t.$set(r)},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function $n(e){let t;const n=e[15].error,r=N(n,e,e[25],rn);return{c(){r&&r.c()},m(e,n){r&&r.m(e,n),t=!0},p(e,s){r&&r.p&&(!t||33554432&s)&&E(r,n,e,e[25],t?A(n,e[25],s,nn):T(e[25]),rn)},i(e){t||(l(r,e),t=!0)},o(e){i(r,e),t=!1},d(e){r&&r.d(e)}}}function hn(e){let t,n,r,s;return{c(){t=d("img"),u(t,"class","pipen-report-image-lg"),F(t.src,n=e[3])||u(t,"src",n),u(t,"alt",e[4])},m(e,n){_(e,t,n),s=!0},p(r,a){e=r,(!s||8&a&&!F(t.src,n=e[3]))&&u(t,"src",n),(!s||16&a)&&u(t,"alt",e[4])},i(n){s||(n&&j(()=>{s&&(r||(r=O(t,z,{duration:e[7]?xn:0},!0)),r.run(1))}),s=!0)},o(n){n&&(r||(r=O(t,z,{duration:e[7]?xn:0},!1)),r.run(0)),s=!1},d(e){e&&a(t),e&&r&&r.end()}}}function bn(e){let t,n,r,s,o,d;const u=[ln,on],m=[];function g(e,t){return void 0===e[6]?0:1}function C(t){e[22](t)}t=g(e),n=m[t]=u[t](e);let S={modalHeading:e[4],primaryButtonText:"Open in new window ...",secondaryButtonText:"Close",$$slots:{default:[hn]},$$scope:{ctx:e}};return void 0!==e[8]&&(S.open=e[8]),s=new ne({props:S}),x.push(()=>v(s,"open",C)),s.$on("click:button--secondary",e[23]),s.$on("click:button--primary",e[24]),{c(){n.c(),r=b(),h(s.$$.fragment)},m(e,n){m[t].m(e,n),_(e,r,n),$(s,e,n),d=!0},p(e,[a]){let _=t;t=g(e),t===_?m[t].p(e,a):(c(),i(m[_],1,1,()=>{m[_]=null}),p(),n=m[t],n?n.p(e,a):(n=m[t]=u[t](e),n.c()),l(n,1),n.m(r.parentNode,r));const d={};16&a&&(d.modalHeading=e[4]),33554584&a&&(d.$$scope={dirty:a,ctx:e}),!o&&256&a&&(o=!0,d.open=e[8],y(()=>o=!1)),s.$set(d)},i(e){d||(l(n),l(s.$$.fragment,e),d=!0)},o(e){i(n),i(s.$$.fragment,e),d=!1},d(e){e&&a(r),m[t].d(e),f(s,e)}}}const xn=110;function vn(e,t,n){const r=["src","alt","download","ratio","loading","loaded","error","fadeIn","border","loadImage"];let s=k(t,r),{$$slots:a={},$$scope:o}=t,{src:i=""}=t,{alt:l=""}=t,{download:c=[]}=t,{ratio:p}=t,{loading:_=!1}=t,{loaded:d=!1}=t,{error:u=!1}=t,{fadeIn:g=!1}=t,{border:f=!0}=t,$=!1,h=!1;const b=e=>{null!=y&&(y=null),n(0,d=!1),n(1,u=!1),y=new Image,y.src=e||i,y.onload=()=>n(0,d=!0),y.onerror=()=>n(1,u=!0)},x=V();let v,y=null;m(()=>()=>y=null);const C={...s,class:[s.class,"pipen-report-image",f?"":"pipen-report-image-noborder"].filter(Boolean).join(" ")};return e.$$set=e=>{t=I(I({},t),B(e)),n(28,s=k(t,r)),"src"in e&&n(3,i=e.src),"alt"in e&&n(4,l=e.alt),"download"in e&&n(5,c=e.download),"ratio"in e&&n(6,p=e.ratio),"loading"in e&&n(2,_=e.loading),"loaded"in e&&n(0,d=e.loaded),"error"in e&&n(1,u=e.error),"fadeIn"in e&&n(7,g=e.fadeIn),"border"in e&&n(13,f=e.border),"$$scope"in e&&n(25,o=e.$$scope)},e.$$.update=()=>{3&e.$$.dirty&&n(2,_=!d&&!u),8&e.$$.dirty&&i&&"undefined"!=typeof window&&b(),1&e.$$.dirty&&d&&x("load"),2&e.$$.dirty&&u&&x("error")},[d,u,_,i,l,c,p,g,$,h,v,()=>{n(8,$=!0)},C,f,b,a,()=>{n(9,h=!0)},()=>{n(10,v=setTimeout(()=>{n(9,h=!1)},100))},()=>{clearTimeout(v),n(9,h=!0)},()=>{n(9,h=!0)},()=>{n(10,v=setTimeout(()=>{n(9,h=!1)},100))},()=>{clearTimeout(v),n(9,h=!0)},function(e){$=e,n(8,$)},()=>{n(8,$=!1)},()=>{n(8,$=!1),window.open(i)},o]}class yn extends t{constructor(e){super(),n(this,e,vn,bn,r,{src:3,alt:4,download:5,ratio:6,loading:2,loaded:0,error:1,fadeIn:7,border:13,loadImage:14})}get loadImage(){return this.$$.ctx[14]}}function Cn(e){let t,n,r=[e[2]],s={};for(let e=0;e<r.length;e+=1)s=I(s,r[e]);return{c(){t=d("div"),n=R("Image loading ..."),D(t,s)},m(e,r){_(e,t,r),C(t,n)},p:M,d(e){e&&a(t)}}}function Sn(e){let t,n,r=[e[3]],s={};for(let e=0;e<r.length;e+=1)s=I(s,r[e]);return{c(){t=d("div"),n=R("Image loading error!"),D(t,s)},m(e,r){_(e,t,r),C(t,n)},p:M,d(e){e&&a(t)}}}function wn(e){let t,n,r;const s=[{src:e[1]},e[4]];let o={$$slots:{error:[Sn],loading:[Cn]},$$scope:{ctx:e}};for(let e=0;e<s.length;e+=1)o=I(o,s[e]);n=new yn({props:o});let c=[e[0]],p={};for(let e=0;e<c.length;e+=1)p=I(p,c[e]);return{c(){t=d("div"),h(n.$$.fragment),D(t,p)},m(e,s){_(e,t,s),$(n,t,null),r=!0},p(e,[r]){const a=18&r?U(s,[2&r&&{src:e[1]},16&r&&P(e[4])]):{};256&r&&(a.$$scope={dirty:r,ctx:e}),n.$set(a),D(t,p=U(c,[1&r&&e[0]]))},i(e){r||(l(n.$$.fragment,e),r=!0)},o(e){i(n.$$.fragment,e),r=!1},d(e){e&&a(t),f(n)}}}function Rn(e,t,n){const r=["frameProps","width","height","src"];let s=k(t,r),{frameProps:a={class:"pipen-report-image-frame"}}=t,{width:o=300}=t,{height:i=200}=t,{src:l}=t;a||(a={"style:position":"relative"});let c=`width: ${o}px; height: ${i}px; aspect-ratio: ${o} / ${i}`,p={...s,class:[s.class,"pipen-report-image-loading"].filter(Boolean).join(" "),style:s.style?`${s.style}; ${c}`:c},_={...p,class:[p.class,"error"].join(" ")};return e.$$set=e=>{t=I(I({},t),B(e)),n(4,s=k(t,r)),"frameProps"in e&&n(0,a=e.frameProps),"width"in e&&n(5,o=e.width),"height"in e&&n(6,i=e.height),"src"in e&&n(1,l=e.src)},[a,l,p,_,s,o,i]}let Mn=class extends t{constructor(e){super(),n(this,e,Rn,wn,r,{frameProps:0,width:5,height:6,src:1})}};function Ln(e){let t,n;return{c(){t=d("div"),n=R(e[0]),u(t,"class","pipen-report-descr-title svelte-1xzxuva")},m(e,r){_(e,t,r),C(t,n)},p(e,t){1&t&&w(n,e[0])},d(e){e&&a(t)}}}function Nn(e){let t,n,r,s=e[0]&&Ln(e);const o=e[3].default,c=N(o,e,e[2],null);let p=[e[1]],u={};for(let e=0;e<p.length;e+=1)u=I(u,p[e]);return{c(){t=d("div"),s&&s.c(),n=b(),c&&c.c(),D(t,u),S(t,"svelte-1xzxuva",!0)},m(e,a){_(e,t,a),s&&s.m(t,null),C(t,n),c&&c.m(t,null),r=!0},p(e,[a]){e[0]?s?s.p(e,a):(s=Ln(e),s.c(),s.m(t,n)):s&&(s.d(1),s=null),c&&c.p&&(!r||4&a)&&E(c,o,e,e[2],r?A(o,e[2],a,null):T(e[2]),null),D(t,u=U(p,[2&a&&e[1]])),S(t,"svelte-1xzxuva",!0)},i(e){r||(l(c,e),r=!0)},o(e){i(c,e),r=!1},d(e){e&&a(t),s&&s.d(),c&&c.d(e)}}}function En(e,t,n){const r=["title"];let s=k(t,r),{$$slots:a={},$$scope:o}=t,{title:i}=t,{klass:l,...c}=s;return l=l?`pipen-report-descr ${l}`:"pipen-report-descr",c||(c={}),c.class=l,e.$$set=e=>{t=I(I({},t),B(e)),n(5,s=k(t,r)),"title"in e&&n(0,i=e.title),"$$scope"in e&&n(2,o=e.$$scope)},[i,c,o,a]}class Tn extends t{constructor(e){super(),n(this,e,En,Nn,r,{title:0})}}function An(e){let t;return{c(){t=R("Dimensionality reduction plot for seurat_clusters")},m(e,n){_(e,t,n)},d(e){e&&a(t)}}}function In(e){let t;return{c(){t=R("Dimensionality reduction plot for VDJ_Presence")},m(e,n){_(e,t,n)},d(e){e&&a(t)}}}function Un(e){let t;return{c(){t=R("The bar plot shows the distribution of cells across categories defined by 'seurat_clusters'. The values represent the number of cells.")},m(e,n){_(e,t,n)},d(e){e&&a(t)}}}function Pn(e){let t;return{c(){t=R("The bar plot shows the distribution of cells across categories defined by 'seurat_clusters', grouped by Sample. The values represent the number of cells.")},m(e,n){_(e,t,n)},d(e){e&&a(t)}}}function kn(e){let t;return{c(){t=R("Number of genes expressed in each cluster")},m(e,n){_(e,t,n)},d(e){e&&a(t)}}}function Bn(e){let t,n,r,s,o,c,p,m,g,x,v,y,C,S,w,R,M,L,N,E,T,A,I,U,P,k,B,V,G,D,O,z,j,F,H;return s=new Tn({props:{$$slots:{default:[An]},$$scope:{ctx:e}}}),c=new Mn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/dimplots/Dimensional-reduction-plot.dim.png",download:[{src:"../SeuratClusterStats/samples.scRep.cluster_stats/dimplots/Dimensional-reduction-plot.dim.pdf"}],width:650,height:550}}),v=new Tn({props:{$$slots:{default:[In]},$$scope:{ctx:e}}}),C=new Mn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/dimplots/VDJ-Presence.dim.png",download:[{src:"../SeuratClusterStats/samples.scRep.cluster_stats/dimplots/VDJ-Presence.dim.pdf"}],width:650,height:550}}),L=new Tn({props:{$$slots:{default:[Un]},$$scope:{ctx:e}}}),E=new Mn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/stats/Number-of-cells-in-each-cluster-Bar-Chart-.png",width:600,height:450}}),P=new Tn({props:{$$slots:{default:[Pn]},$$scope:{ctx:e}}}),B=new Mn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/stats/Number-of-cells-in-each-cluster-by-Sample-Bar-Chart-.png",width:690,height:450}}),z=new Tn({props:{$$slots:{default:[kn]},$$scope:{ctx:e}}}),F=new Mn({props:{src:"../SeuratClusterStats/samples.scRep.cluster_stats/ngenes/Number-of-genes-expressed-in-each-cluster.png",width:1e3,height:800}}),{c(){t=d("h1"),t.textContent="Dimensional reduction plot",n=d("a"),n.innerHTML="",r=b(),h(s.$$.fragment),o=b(),h(c.$$.fragment),p=b(),m=d("h1"),m.textContent="VDJ Presence",g=d("a"),g.innerHTML="",x=b(),h(v.$$.fragment),y=b(),h(C.$$.fragment),S=b(),w=d("h1"),w.textContent="Number of cells in each cluster (Bar Chart)",R=d("a"),R.innerHTML="",M=b(),h(L.$$.fragment),N=b(),h(E.$$.fragment),T=b(),A=d("h1"),A.textContent="Number of cells in each cluster by Sample (Bar Chart)",I=d("a"),I.innerHTML="",U=b(),h(P.$$.fragment),k=b(),h(B.$$.fragment),V=b(),G=d("h1"),G.textContent="Number of genes expressed in each cluster",D=d("a"),D.innerHTML="",O=b(),h(z.$$.fragment),j=b(),h(F.$$.fragment),u(n,"id","prt-h1-0-dimensional-reduction-plot"),u(n,"class","pipen-report-toc-anchor"),u(g,"id","prt-h1-1-vdj-presence"),u(g,"class","pipen-report-toc-anchor"),u(R,"id","prt-h1-2-number-of-cells-in-each-cluster-bar-chart"),u(R,"class","pipen-report-toc-anchor"),u(I,"id","prt-h1-3-number-of-cells-in-each-cluster-by-sample-bar-chart"),u(I,"class","pipen-report-toc-anchor"),u(D,"id","prt-h1-4-number-of-genes-expressed-in-each-cluster"),u(D,"class","pipen-report-toc-anchor")},m(e,a){_(e,t,a),_(e,n,a),_(e,r,a),$(s,e,a),_(e,o,a),$(c,e,a),_(e,p,a),_(e,m,a),_(e,g,a),_(e,x,a),$(v,e,a),_(e,y,a),$(C,e,a),_(e,S,a),_(e,w,a),_(e,R,a),_(e,M,a),$(L,e,a),_(e,N,a),$(E,e,a),_(e,T,a),_(e,A,a),_(e,I,a),_(e,U,a),$(P,e,a),_(e,k,a),$(B,e,a),_(e,V,a),_(e,G,a),_(e,D,a),_(e,O,a),$(z,e,a),_(e,j,a),$(F,e,a),H=!0},p(e,[t]){const n={};1&t&&(n.$$scope={dirty:t,ctx:e}),s.$set(n);const r={};1&t&&(r.$$scope={dirty:t,ctx:e}),v.$set(r);const a={};1&t&&(a.$$scope={dirty:t,ctx:e}),L.$set(a);const o={};1&t&&(o.$$scope={dirty:t,ctx:e}),P.$set(o);const i={};1&t&&(i.$$scope={dirty:t,ctx:e}),z.$set(i)},i(e){H||(l(s.$$.fragment,e),l(c.$$.fragment,e),l(v.$$.fragment,e),l(C.$$.fragment,e),l(L.$$.fragment,e),l(E.$$.fragment,e),l(P.$$.fragment,e),l(B.$$.fragment,e),l(z.$$.fragment,e),l(F.$$.fragment,e),H=!0)},o(e){i(s.$$.fragment,e),i(c.$$.fragment,e),i(v.$$.fragment,e),i(C.$$.fragment,e),i(L.$$.fragment,e),i(E.$$.fragment,e),i(P.$$.fragment,e),i(B.$$.fragment,e),i(z.$$.fragment,e),i(F.$$.fragment,e),H=!1},d(e){e&&(a(t),a(n),a(r),a(o),a(p),a(m),a(g),a(x),a(y),a(S),a(w),a(R),a(M),a(N),a(T),a(A),a(I),a(U),a(k),a(V),a(G),a(D),a(O),a(j)),f(s,e),f(c,e),f(v,e),f(C,e),f(L,e),f(E,e),f(P,e),f(B,e),f(z,e),f(F,e)}}}class Vn extends t{constructor(e){super(),n(this,e,null,Bn,r,{})}}var Gn=[{slug:"prt-h1-0-dimensional-reduction-plot",text:"Dimensional reduction plot",children:[],page:0},{slug:"prt-h1-1-vdj-presence",text:"VDJ Presence",children:[],page:0},{slug:"prt-h1-2-number-of-cells-in-each-cluster-bar-chart",text:"Number of cells in each cluster (Bar Chart)",children:[],page:0},{slug:"prt-h1-3-number-of-cells-in-each-cluster-by-sample-bar-chart",text:"Number of cells in each cluster by Sample (Bar Chart)",children:[],page:0},{slug:"prt-h1-4-number-of-genes-expressed-in-each-cluster",text:"Number of genes expressed in each cluster",children:[],page:0}],Dn={pipeline:{name:"Immunopipe"},versions:'\n    <a href="https://github.com/pwwang/pipen" target="_blank">\n        pipen v1.1.8\n    </a> and\n    <a href="https://github.com/pwwang/pipen-report" target="_blank">\n        pipen-report v1.1.0\n    </a>\n',entries:[{name:"SampleInfo",desc:"List sample information and perform statistics",npages:1,report_toc:!0,order:0,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] plotthis_0.10.0      biopipen.utils_0.3.5 gglogger_0.1.7      \n[4] dplyr_1.1.4          rlang_1.1.6         \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                tibble_3.3.0               \n  [7] polyclip_1.10-7             enrichit_0.0.4             \n  [9] fastDummies_1.7.5           lifecycle_1.0.4            \n [11] globals_0.18.0              lattice_0.22-7             \n [13] MASS_7.3-60.0.1             magrittr_2.0.3             \n [15] plotly_4.11.0               httpuv_1.6.16              \n [17] Seurat_5.3.0                sctransform_0.4.2          \n [19] spam_2.11-1                 sp_2.2-0                   \n [21] spatstat.sparse_3.1-0       reticulate_1.43.0          \n [23] cowplot_1.2.0               pbapply_1.7-4              \n [25] RColorBrewer_1.1-3          abind_1.4-5                \n [27] zlibbioc_1.48.0             Rtsne_0.17                 \n [29] GenomicRanges_1.54.1        purrr_1.1.0                \n [31] ggraph_2.2.1                BiocGenerics_0.48.1        \n [33] RCurl_1.98-1.17             tweenr_2.0.3               \n [35] evmix_2.12                  circlize_0.4.16            \n [37] GenomeInfoDbData_1.2.11     IRanges_2.36.0             \n [39] S4Vectors_0.40.2            ggrepel_0.9.6              \n [41] irlba_2.3.5.1               listenv_0.9.1              \n [43] spatstat.utils_3.1-5        iNEXT_3.0.2                \n [45] MatrixModels_0.5-4          goftest_1.2-3              \n [47] RSpectra_0.16-2             scRepertoire_2.2.1         \n [49] spatstat.random_3.4-1       fitdistrplus_1.2-4         \n [51] parallelly_1.45.1           codetools_0.2-20           \n [53] DelayedArray_0.28.0         xml2_1.4.0                 \n [55] ggforce_0.5.0               tidyselect_1.2.1           \n [57] shape_1.4.6.1               farver_2.1.2               \n [59] viridis_0.6.5               matrixStats_1.5.0          \n [61] stats4_4.3.3                spatstat.explore_3.5-2     \n [63] jsonlite_2.0.0              ellipsis_0.3.2             \n [65] tidygraph_1.3.0             progressr_0.15.1           \n [67] ggridges_0.5.7              ggalluvial_0.12.5          \n [69] survival_3.8-3              tools_4.3.3                \n [71] ggnewscale_0.5.2            stringdist_0.9.15          \n [73] ica_1.0-3                   Rcpp_1.1.0                 \n [75] glue_1.8.0                  gridExtra_2.3              \n [77] SparseArray_1.2.2           qvalue_2.34.0              \n [79] MatrixGenerics_1.14.0       GenomeInfoDb_1.38.1        \n [81] withr_3.0.2                 fastmap_1.2.0              \n [83] fansi_1.0.6                 SparseM_1.84-2             \n [85] ttservice_0.5.3             digest_0.6.37              \n [87] R6_2.6.1                    mime_0.13                  \n [89] colorspace_2.1-1            scattermore_1.2            \n [91] tensor_1.5.1                dichromat_2.0-0.1          \n [93] spatstat.data_3.1-8         tidyr_1.3.1                \n [95] generics_0.1.4              data.table_1.17.8          \n [97] graphlayouts_1.2.2          httr_1.4.7                 \n [99] htmlwidgets_1.6.4           S4Arrays_1.2.0             \n[101] uwot_0.2.3                  pkgconfig_2.0.3            \n[103] gtable_0.3.6                lmtest_0.9-40              \n[105] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[107] htmltools_0.5.8.1           dotCall64_1.2              \n[109] SeuratObject_5.2.0          scales_1.4.0               \n[111] Biobase_2.62.0              png_0.1-8                  \n[113] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[115] rjson_0.2.23                reshape2_1.4.4             \n[117] nlme_3.1-168                cachem_1.1.0               \n[119] zoo_1.8-14                  GlobalOptions_0.1.2        \n[121] stringr_1.5.2               KernSmooth_2.23-26         \n[123] parallel_4.3.3              miniUI_0.1.2               \n[125] pillar_1.11.0               grid_4.3.3                 \n[127] logger_0.4.0                vctrs_0.6.5                \n[129] RANN_2.6.2                  VGAM_1.1-13                \n[131] promises_1.3.3              xtable_1.8-4               \n[133] cluster_2.1.8.1             truncdist_1.0-2            \n[135] cli_3.6.5                   compiler_4.3.3             \n[137] crayon_1.5.3                future.apply_1.20.0        \n[139] plyr_1.8.9                  forcats_1.0.0              \n[141] stringi_1.8.7               viridisLite_0.4.2          \n[143] tidyseurat_0.8.0            deldir_2.0-4               \n[145] scplotter_0.6.5             assertthat_0.2.1           \n[147] gsl_2.1-8                   lazyeval_0.2.2             \n[149] spatstat.geom_3.5-0         quantreg_6.1               \n[151] Matrix_1.6-5                RcppHNSW_0.6.0             \n[153] patchwork_1.3.2             future_1.67.0              \n[155] ggplot2_3.5.2               shiny_1.11.1               \n[157] SummarizedExperiment_1.32.0 evd_2.3-7.1                \n[159] ROCR_1.0-11                 gridtext_0.1.5             \n[161] memoise_2.0.1               igraph_2.1.4               \n[163] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/SampleInfo/0/job.script\nVoluntary context switches: 356\nInvoluntary context switches: 110\nPercentage of CPU this job got: 160%\nMajor page faults: 0\nMinor page faults: 202689\nMaximum resident set size (kB): 813008\nElapsed real time (s): 8.78\nSystem (kernel) time (s): 0.50\nUser time (s): 13.57\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\n"}},{name:"SeuratPreparing",desc:"Load, prepare and apply QC to data, using `Seurat`",npages:1,report_toc:!0,order:2,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n[1] biopipen.utils_0.3.5 glue_1.8.0           dplyr_1.1.4         \n[4] bracer_1.2.1         future_1.67.0        Seurat_5.3.0        \n[7] SeuratObject_5.2.0   sp_2.2-0            \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                R.oo_1.27.1                \n  [7] tibble_3.3.0                polyclip_1.10-7            \n  [9] enrichit_0.0.4              fastDummies_1.7.5          \n [11] lifecycle_1.0.4             globals_0.18.0             \n [13] lattice_0.22-7              MASS_7.3-60.0.1            \n [15] magrittr_2.0.3              plotly_4.11.0              \n [17] httpuv_1.6.16               sctransform_0.4.2          \n [19] spam_2.11-1                 spatstat.sparse_3.1-0      \n [21] reticulate_1.43.0           cowplot_1.2.0              \n [23] pbapply_1.7-4               RColorBrewer_1.1-3         \n [25] abind_1.4-5                 zlibbioc_1.48.0            \n [27] Rtsne_0.17                  GenomicRanges_1.54.1       \n [29] R.utils_2.13.0              purrr_1.1.0                \n [31] ggraph_2.2.1                BiocGenerics_0.48.1        \n [33] RCurl_1.98-1.17             tweenr_2.0.3               \n [35] evmix_2.12                  circlize_0.4.16            \n [37] GenomeInfoDbData_1.2.11     IRanges_2.36.0             \n [39] S4Vectors_0.40.2            ggrepel_0.9.6              \n [41] irlba_2.3.5.1               listenv_0.9.1              \n [43] spatstat.utils_3.1-5        iNEXT_3.0.2                \n [45] MatrixModels_0.5-4          goftest_1.2-3              \n [47] RSpectra_0.16-2             scRepertoire_2.2.1         \n [49] spatstat.random_3.4-1       fitdistrplus_1.2-4         \n [51] parallelly_1.45.1           codetools_0.2-20           \n [53] DelayedArray_0.28.0         xml2_1.4.0                 \n [55] ggforce_0.5.0               tidyselect_1.2.1           \n [57] shape_1.4.6.1               farver_2.1.2               \n [59] viridis_0.6.5               matrixStats_1.5.0          \n [61] stats4_4.3.3                spatstat.explore_3.5-2     \n [63] jsonlite_2.0.0              ellipsis_0.3.2             \n [65] tidygraph_1.3.0             progressr_0.15.1           \n [67] ggridges_0.5.7              ggalluvial_0.12.5          \n [69] survival_3.8-3              systemfonts_1.2.3          \n [71] tools_4.3.3                 ggnewscale_0.5.2           \n [73] ragg_1.5.0                  stringdist_0.9.15          \n [75] ica_1.0-3                   Rcpp_1.1.0                 \n [77] gridExtra_2.3               SparseArray_1.2.2          \n [79] qs2_0.1.5                   qvalue_2.34.0              \n [81] MatrixGenerics_1.14.0       GenomeInfoDb_1.38.1        \n [83] withr_3.0.2                 fastmap_1.2.0              \n [85] fansi_1.0.6                 SparseM_1.84-2             \n [87] ttservice_0.5.3             digest_0.6.37              \n [89] R6_2.6.1                    mime_0.13                  \n [91] textshaping_0.4.0           colorspace_2.1-1           \n [93] scattermore_1.2             tensor_1.5.1               \n [95] dichromat_2.0-0.1           spatstat.data_3.1-8        \n [97] R.methodsS3_1.8.2           RhpcBLASctl_0.23-42        \n [99] tidyr_1.3.1                 generics_0.1.4             \n[101] data.table_1.17.8           graphlayouts_1.2.2         \n[103] httr_1.4.7                  htmlwidgets_1.6.4          \n[105] S4Arrays_1.2.0              uwot_0.2.3                 \n[107] pkgconfig_2.0.3             gtable_0.3.6               \n[109] lmtest_0.9-40               SingleCellExperiment_1.24.0\n[111] XVector_0.42.0              htmltools_0.5.8.1          \n[113] dotCall64_1.2               gglogger_0.1.7             \n[115] scales_1.4.0                Biobase_2.62.0             \n[117] png_0.1-8                   harmony_1.2.0              \n[119] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[121] rjson_0.2.23                reshape2_1.4.4             \n[123] nlme_3.1-168                cachem_1.1.0               \n[125] zoo_1.8-14                  GlobalOptions_0.1.2        \n[127] stringr_1.5.2               KernSmooth_2.23-26         \n[129] parallel_4.3.3              miniUI_0.1.2               \n[131] pillar_1.11.0               grid_4.3.3                 \n[133] logger_0.4.0                vctrs_0.6.5                \n[135] RANN_2.6.2                  VGAM_1.1-13                \n[137] promises_1.3.3              stringfish_0.17.0          \n[139] xtable_1.8-4                cluster_2.1.8.1            \n[141] truncdist_1.0-2             cli_3.6.5                  \n[143] compiler_4.3.3              rlang_1.1.6                \n[145] crayon_1.5.3                future.apply_1.20.0        \n[147] labeling_0.4.3              plyr_1.8.9                 \n[149] forcats_1.0.0               stringi_1.8.7              \n[151] viridisLite_0.4.2           tidyseurat_0.8.0           \n[153] deldir_2.0-4                scplotter_0.6.5            \n[155] assertthat_0.2.1            gsl_2.1-8                  \n[157] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[159] quantreg_6.1                Matrix_1.6-5               \n[161] RcppHNSW_0.6.0              patchwork_1.3.2            \n[163] ggplot2_3.5.2               shiny_1.11.1               \n[165] plotthis_0.10.0             SummarizedExperiment_1.32.0\n[167] evd_2.3-7.1                 ROCR_1.0-11                \n[169] gridtext_0.1.5              memoise_2.0.1              \n[171] igraph_2.1.4                RcppParallel_5.1.9         \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/SeuratPreparing/0/job.script\nVoluntary context switches: 3677\nInvoluntary context switches: 409\nPercentage of CPU this job got: 155%\nMajor page faults: 0\nMinor page faults: 1929090\nMaximum resident set size (kB): 3128452\nElapsed real time (s): 56.75\nSystem (kernel) time (s): 2.82\nUser time (s): 85.55\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\n4807a2eabf37\n\nCPU\n----\nArchitecture:                            x86_64\nCPU op-mode(s):                          32-bit, 64-bit\nAddress sizes:                           46 bits physical, 48 bits virtual\nByte Order:                              Little Endian\nCPU(s):                                  64\nOn-line CPU(s) list:                     0-63\nVendor ID:                               GenuineIntel\nModel name:                              Intel(R) Xeon(R) CPU @ 2.80GHz\nCPU family:                              6\nModel:                                   85\nThread(s) per core:                      2\nCore(s) per socket:                      16\nSocket(s):                               2\nStepping:                                7\nBogoMIPS:                                5599.99\nFlags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat avx512_vnni md_clear arch_capabilities\nHypervisor vendor:                       KVM\nVirtualization type:                     full\nL1d cache:                               1 MiB (32 instances)\nL1i cache:                               1 MiB (32 instances)\nL2 cache:                                32 MiB (32 instances)\nL3 cache:                                66 MiB (2 instances)\nNUMA node(s):                            2\nNUMA node0 CPU(s):                       0-15,32-47\nNUMA node1 CPU(s):                       16-31,48-63\nVulnerability Gather data sampling:      Not affected\nVulnerability Indirect target selection: Mitigation; Aligned branch/return thunks\nVulnerability Itlb multihit:             Not affected\nVulnerability L1tf:                      Not affected\nVulnerability Mds:                       Not affected\nVulnerability Meltdown:                  Not affected\nVulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Reg file data sampling:    Not affected\nVulnerability Retbleed:                  Mitigation; Enhanced IBRS\nVulnerability Spec rstack overflow:      Not affected\nVulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                     Not affected\nVulnerability Tsa:                       Not affected\nVulnerability Tsx async abort:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Vmscape:                   Not affected\n\nMemory\n"}},{name:"ClusterMarkersOfAllCells",desc:"Markers for clusters of all cells.",npages:1,report_toc:!0,order:4,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] biopipen.utils_0.3.5 plotthis_0.10.0      ggplot2_3.5.2       \n [4] tidyr_1.3.1          tidyseurat_0.8.0     ttservice_0.5.3     \n [7] Seurat_5.3.0         SeuratObject_5.2.0   sp_2.2-0            \n[10] dplyr_1.1.4          rlang_1.1.6         \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                tibble_3.3.0               \n  [7] polyclip_1.10-7             enrichit_0.0.4             \n  [9] fastDummies_1.7.5           lifecycle_1.0.4            \n [11] doParallel_1.0.17           globals_0.18.0             \n [13] lattice_0.22-7              MASS_7.3-60.0.1            \n [15] magrittr_2.0.3              limma_3.58.1               \n [17] plotly_4.11.0               httpuv_1.6.16              \n [19] sctransform_0.4.2           spam_2.11-1                \n [21] spatstat.sparse_3.1-0       reticulate_1.43.0          \n [23] cowplot_1.2.0               pbapply_1.7-4              \n [25] RColorBrewer_1.1-3          abind_1.4-5                \n [27] zlibbioc_1.48.0             Rtsne_0.17                 \n [29] GenomicRanges_1.54.1        presto_1.0.0               \n [31] purrr_1.1.0                 ggraph_2.2.1               \n [33] BiocGenerics_0.48.1         RCurl_1.98-1.17            \n [35] tweenr_2.0.3                evmix_2.12                 \n [37] circlize_0.4.16             GenomeInfoDbData_1.2.11    \n [39] IRanges_2.36.0              S4Vectors_0.40.2           \n [41] ggrepel_0.9.6               irlba_2.3.5.1              \n [43] listenv_0.9.1               spatstat.utils_3.1-5       \n [45] iNEXT_3.0.2                 MatrixModels_0.5-4         \n [47] goftest_1.2-3               RSpectra_0.16-2            \n [49] scRepertoire_2.2.1          spatstat.random_3.4-1      \n [51] fitdistrplus_1.2-4          parallelly_1.45.1          \n [53] codetools_0.2-20            DelayedArray_0.28.0        \n [55] xml2_1.4.0                  ggforce_0.5.0              \n [57] tidyselect_1.2.1            shape_1.4.6.1              \n [59] farver_2.1.2                viridis_0.6.5              \n [61] matrixStats_1.5.0           stats4_4.3.3               \n [63] spatstat.explore_3.5-2      jsonlite_2.0.0             \n [65] GetoptLong_1.0.5            ellipsis_0.3.2             \n [67] tidygraph_1.3.0             progressr_0.15.1           \n [69] iterators_1.0.14            ggridges_0.5.7             \n [71] ggalluvial_0.12.5           survival_3.8-3             \n [73] systemfonts_1.2.3           foreach_1.5.2              \n [75] tools_4.3.3                 ggnewscale_0.5.2           \n [77] ragg_1.5.0                  stringdist_0.9.15          \n [79] ica_1.0-3                   Rcpp_1.1.0                 \n [81] glue_1.8.0                  gridExtra_2.3              \n [83] SparseArray_1.2.2           qs2_0.1.5                  \n [85] qvalue_2.34.0               MatrixGenerics_1.14.0      \n [87] GenomeInfoDb_1.38.1         withr_3.0.2                \n [89] fastmap_1.2.0               fansi_1.0.6                \n [91] SparseM_1.84-2              digest_0.6.37              \n [93] R6_2.6.1                    mime_0.13                  \n [95] textshaping_0.4.0           colorspace_2.1-1           \n [97] Cairo_1.6-2                 scattermore_1.2            \n [99] tensor_1.5.1                dichromat_2.0-0.1          \n[101] spatstat.data_3.1-8         generics_0.1.4             \n[103] data.table_1.17.8           graphlayouts_1.2.2         \n[105] httr_1.4.7                  htmlwidgets_1.6.4          \n[107] S4Arrays_1.2.0              uwot_0.2.3                 \n[109] pkgconfig_2.0.3             gtable_0.3.6               \n[111] ComplexHeatmap_2.18.0       lmtest_0.9-40              \n[113] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[115] htmltools_0.5.8.1           dotCall64_1.2              \n[117] clue_0.3-66                 scales_1.4.0               \n[119] Biobase_2.62.0              png_0.1-8                  \n[121] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[123] rjson_0.2.23                reshape2_1.4.4             \n[125] nlme_3.1-168                cachem_1.1.0               \n[127] zoo_1.8-14                  GlobalOptions_0.1.2        \n[129] stringr_1.5.2               KernSmooth_2.23-26         \n[131] parallel_4.3.3              miniUI_0.1.2               \n[133] pillar_1.11.0               grid_4.3.3                 \n[135] logger_0.4.0                vctrs_0.6.5                \n[137] RANN_2.6.2                  VGAM_1.1-13                \n[139] promises_1.3.3              stringfish_0.17.0          \n[141] xtable_1.8-4                cluster_2.1.8.1            \n[143] truncdist_1.0-2             cli_3.6.5                  \n[145] compiler_4.3.3              crayon_1.5.3               \n[147] future.apply_1.20.0         labeling_0.4.3             \n[149] plyr_1.8.9                  forcats_1.0.0              \n[151] stringi_1.8.7               viridisLite_0.4.2          \n[153] deldir_2.0-4                scplotter_0.6.5            \n[155] assertthat_0.2.1            gsl_2.1-8                  \n[157] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[159] quantreg_6.1                Matrix_1.6-5               \n[161] RcppHNSW_0.6.0              patchwork_1.3.2            \n[163] future_1.67.0               statmod_1.5.0              \n[165] shiny_1.11.1                SummarizedExperiment_1.32.0\n[167] evd_2.3-7.1                 ROCR_1.0-11                \n[169] gridtext_0.1.5              memoise_2.0.1              \n[171] igraph_2.1.4                RcppParallel_5.1.9         \n[173] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/ClusterMarkersOfAllCells/0/job.script\nVoluntary context switches: 10326\nInvoluntary context switches: 529\nPercentage of CPU this job got: 89%\nMajor page faults: 0\nMinor page faults: 9978755\nMaximum resident set size (kB): 3005976\nElapsed real time (s): 143.75\nSystem (kernel) time (s): 12.21\nUser time (s): 116.90\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\n4807a2eabf37\n\nCPU\n----\nArchitecture:                            x86_64\nCPU op-mode(s):                          32-bit, 64-bit\nAddress sizes:                           46 bits physical, 48 bits virtual\nByte Order:                              Little Endian\nCPU(s):                                  64\nOn-line CPU(s) list:                     0-63\nVendor ID:                               GenuineIntel\nModel name:                              Intel(R) Xeon(R) CPU @ 2.80GHz\nCPU family:                              6\nModel:                                   85\nThread(s) per core:                      2\nCore(s) per socket:                      16\nSocket(s):                               2\nStepping:                                7\nBogoMIPS:                                5599.99\nFlags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 hle avx2 smep bmi2 erms invpcid rtm avx512f avx512dq rdseed adx smap clflushopt clwb avx512cd avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves arat avx512_vnni md_clear arch_capabilities\nHypervisor vendor:                       KVM\nVirtualization type:                     full\nL1d cache:                               1 MiB (32 instances)\nL1i cache:                               1 MiB (32 instances)\nL2 cache:                                32 MiB (32 instances)\nL3 cache:                                66 MiB (2 instances)\nNUMA node(s):                            2\nNUMA node0 CPU(s):                       0-15,32-47\nNUMA node1 CPU(s):                       16-31,48-63\nVulnerability Gather data sampling:      Not affected\nVulnerability Indirect target selection: Mitigation; Aligned branch/return thunks\nVulnerability Itlb multihit:             Not affected\nVulnerability L1tf:                      Not affected\nVulnerability Mds:                       Not affected\nVulnerability Meltdown:                  Not affected\nVulnerability Mmio stale data:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Reg file data sampling:    Not affected\nVulnerability Retbleed:                  Mitigation; Enhanced IBRS\nVulnerability Spec rstack overflow:      Not affected\nVulnerability Spec store bypass:         Mitigation; Speculative Store Bypass disabled via prctl\nVulnerability Spectre v1:                Mitigation; usercopy/swapgs barriers and __user pointer sanitization\nVulnerability Spectre v2:                Mitigation; Enhanced / Automatic IBRS; IBPB conditional; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop\nVulnerability Srbds:                     Not affected\nVulnerability Tsa:                       Not affected\nVulnerability Tsx async abort:           Vulnerable: Clear CPU buffers attempted, no microcode; SMT Host state unknown\nVulnerability Vmscape:                   Not affected\n\n"}},{name:"TOrBCellSelection",desc:"Separate T and non-T cells and select T cells; or separate B and non-B cells and select B cells.",npages:1,report_toc:!0,order:5,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] biopipen.utils_0.3.5 scplotter_0.6.5      plotthis_0.10.0     \n [4] Seurat_5.3.0         SeuratObject_5.2.0   sp_2.2-0            \n [7] tibble_3.3.0         tidyr_1.3.1          dplyr_1.1.4         \n[10] rlang_1.1.6         \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                polyclip_1.10-7            \n  [7] enrichit_0.0.4              fastDummies_1.7.5          \n  [9] lifecycle_1.0.4             globals_0.18.0             \n [11] lattice_0.22-7              MASS_7.3-60.0.1            \n [13] magrittr_2.0.3              plotly_4.11.0              \n [15] httpuv_1.6.16               sctransform_0.4.2          \n [17] spam_2.11-1                 spatstat.sparse_3.1-0      \n [19] reticulate_1.43.0           cowplot_1.2.0              \n [21] pbapply_1.7-4               RColorBrewer_1.1-3         \n [23] abind_1.4-5                 zlibbioc_1.48.0            \n [25] Rtsne_0.17                  GenomicRanges_1.54.1       \n [27] purrr_1.1.0                 ggraph_2.2.1               \n [29] BiocGenerics_0.48.1         RCurl_1.98-1.17            \n [31] tweenr_2.0.3                evmix_2.12                 \n [33] circlize_0.4.16             GenomeInfoDbData_1.2.11    \n [35] IRanges_2.36.0              S4Vectors_0.40.2           \n [37] ggrepel_0.9.6               irlba_2.3.5.1              \n [39] listenv_0.9.1               spatstat.utils_3.1-5       \n [41] iNEXT_3.0.2                 MatrixModels_0.5-4         \n [43] goftest_1.2-3               RSpectra_0.16-2            \n [45] scRepertoire_2.2.1          spatstat.random_3.4-1      \n [47] fitdistrplus_1.2-4          parallelly_1.45.1          \n [49] codetools_0.2-20            DelayedArray_0.28.0        \n [51] xml2_1.4.0                  ggforce_0.5.0              \n [53] tidyselect_1.2.1            shape_1.4.6.1              \n [55] farver_2.1.2                viridis_0.6.5              \n [57] matrixStats_1.5.0           stats4_4.3.3               \n [59] spatstat.explore_3.5-2      jsonlite_2.0.0             \n [61] ellipsis_0.3.2              tidygraph_1.3.0            \n [63] progressr_0.15.1            ggridges_0.5.7             \n [65] ggalluvial_0.12.5           survival_3.8-3             \n [67] tools_4.3.3                 ggnewscale_0.5.2           \n [69] stringdist_0.9.15           ica_1.0-3                  \n [71] Rcpp_1.1.0                  glue_1.8.0                 \n [73] gridExtra_2.3               SparseArray_1.2.2          \n [75] qs2_0.1.5                   qvalue_2.34.0              \n [77] MatrixGenerics_1.14.0       GenomeInfoDb_1.38.1        \n [79] withr_3.0.2                 fastmap_1.2.0              \n [81] fansi_1.0.6                 ttservice_0.5.3            \n [83] SparseM_1.84-2              digest_0.6.37              \n [85] R6_2.6.1                    mime_0.13                  \n [87] colorspace_2.1-1            scattermore_1.2            \n [89] tensor_1.5.1                dichromat_2.0-0.1          \n [91] spatstat.data_3.1-8         generics_0.1.4             \n [93] data.table_1.17.8           graphlayouts_1.2.2         \n [95] httr_1.4.7                  htmlwidgets_1.6.4          \n [97] S4Arrays_1.2.0              uwot_0.2.3                 \n [99] pkgconfig_2.0.3             gtable_0.3.6               \n[101] lmtest_0.9-40               SingleCellExperiment_1.24.0\n[103] XVector_0.42.0              htmltools_0.5.8.1          \n[105] dotCall64_1.2               scales_1.4.0               \n[107] Biobase_2.62.0              png_0.1-8                  \n[109] spatstat.univar_3.1-4       ggdendro_0.2.0             \n[111] reshape2_1.4.4              rjson_0.2.23               \n[113] nlme_3.1-168                zoo_1.8-14                 \n[115] cachem_1.1.0                GlobalOptions_0.1.2        \n[117] stringr_1.5.2               KernSmooth_2.23-26         \n[119] parallel_4.3.3              miniUI_0.1.2               \n[121] logger_0.4.0                pillar_1.11.0              \n[123] grid_4.3.3                  vctrs_0.6.5                \n[125] RANN_2.6.2                  VGAM_1.1-13                \n[127] promises_1.3.3              stringfish_0.17.0          \n[129] xtable_1.8-4                cluster_2.1.8.1            \n[131] truncdist_1.0-2             cli_3.6.5                  \n[133] compiler_4.3.3              crayon_1.5.3               \n[135] future.apply_1.20.0         labeling_0.4.3             \n[137] plyr_1.8.9                  forcats_1.0.0              \n[139] stringi_1.8.7               tidyseurat_0.8.0           \n[141] viridisLite_0.4.2           deldir_2.0-4               \n[143] assertthat_0.2.1            gsl_2.1-8                  \n[145] lazyeval_0.2.2              spatstat.geom_3.5-0        \n[147] quantreg_6.1                Matrix_1.6-5               \n[149] RcppHNSW_0.6.0              patchwork_1.3.2            \n[151] future_1.67.0               ggplot2_3.5.2              \n[153] shiny_1.11.1                SummarizedExperiment_1.32.0\n[155] evd_2.3-7.1                 ROCR_1.0-11                \n[157] gridtext_0.1.5              igraph_2.1.4               \n[159] memoise_2.0.1               RcppParallel_5.1.9         \n[161] bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/TOrBCellSelection/0/job.script\nVoluntary context switches: 988\nInvoluntary context switches: 262\nPercentage of CPU this job got: 116%\nMajor page faults: 0\nMinor page faults: 638542\nMaximum resident set size (kB): 2092052\nElapsed real time (s): 19.48\nSystem (kernel) time (s): 0.95\nUser time (s): 21.65\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\n"}},{name:"SeuratClusterStats",desc:"Statistics of the clustering.",npages:1,report_toc:!0,order:8,runinfo:{session:"# Generated by pipen_runinfo v%(version)s\n# Lang: R\nR version 4.3.3 (2024-02-29)\nPlatform: x86_64-conda-linux-gnu (64-bit)\nRunning under: Debian GNU/Linux 12 (bookworm)\n\nMatrix products: default\nBLAS/LAPACK: /opt/conda/lib/libopenblasp-r0.3.30.so;  LAPACK version 3.12.0\n\nlocale:\n [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       \n [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   \n [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          \n[10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   \n\ntime zone: Etc/UTC\ntzcode source: system (glibc)\n\nattached base packages:\n[1] stats     graphics  grDevices utils     datasets  methods   base     \n\nother attached packages:\n [1] biopipen.utils_0.3.5 scplotter_0.6.5      gglogger_0.1.7      \n [4] ggplot2_3.5.2        tidyseurat_0.8.0     ttservice_0.5.3     \n [7] forcats_1.0.0        glue_1.8.0           tibble_3.3.0        \n[10] tidyr_1.3.1          dplyr_1.1.4          rlang_1.1.6         \n[13] Seurat_5.3.0         SeuratObject_5.2.0   sp_2.2-0            \n\nloaded via a namespace (and not attached):\n  [1] cubature_2.0.4.6            RcppAnnoy_0.0.22           \n  [3] splines_4.3.3               later_1.4.4                \n  [5] bitops_1.0-9                polyclip_1.10-7            \n  [7] enrichit_0.0.4              fastDummies_1.7.5          \n  [9] lifecycle_1.0.4             globals_0.18.0             \n [11] lattice_0.22-7              MASS_7.3-60.0.1            \n [13] magrittr_2.0.3              plotly_4.11.0              \n [15] httpuv_1.6.16               sctransform_0.4.2          \n [17] spam_2.11-1                 spatstat.sparse_3.1-0      \n [19] reticulate_1.43.0           cowplot_1.2.0              \n [21] pbapply_1.7-4               RColorBrewer_1.1-3         \n [23] abind_1.4-5                 zlibbioc_1.48.0            \n [25] Rtsne_0.17                  GenomicRanges_1.54.1       \n [27] purrr_1.1.0                 ggraph_2.2.1               \n [29] BiocGenerics_0.48.1         RCurl_1.98-1.17            \n [31] tweenr_2.0.3                evmix_2.12                 \n [33] circlize_0.4.16             GenomeInfoDbData_1.2.11    \n [35] IRanges_2.36.0              S4Vectors_0.40.2           \n [37] ggrepel_0.9.6               irlba_2.3.5.1              \n [39] listenv_0.9.1               spatstat.utils_3.1-5       \n [41] iNEXT_3.0.2                 MatrixModels_0.5-4         \n [43] goftest_1.2-3               RSpectra_0.16-2            \n [45] scRepertoire_2.2.1          spatstat.random_3.4-1      \n [47] fitdistrplus_1.2-4          parallelly_1.45.1          \n [49] codetools_0.2-20            DelayedArray_0.28.0        \n [51] xml2_1.4.0                  ggforce_0.5.0              \n [53] tidyselect_1.2.1            shape_1.4.6.1              \n [55] farver_2.1.2                viridis_0.6.5              \n [57] matrixStats_1.5.0           stats4_4.3.3               \n [59] spatstat.explore_3.5-2      jsonlite_2.0.0             \n [61] ellipsis_0.3.2              tidygraph_1.3.0            \n [63] progressr_0.15.1            ggridges_0.5.7             \n [65] ggalluvial_0.12.5           survival_3.8-3             \n [67] systemfonts_1.2.3           tools_4.3.3                \n [69] ggnewscale_0.5.2            ragg_1.5.0                 \n [71] stringdist_0.9.15           ica_1.0-3                  \n [73] Rcpp_1.1.0                  gridExtra_2.3              \n [75] SparseArray_1.2.2           qs2_0.1.5                  \n [77] qvalue_2.34.0               MatrixGenerics_1.14.0      \n [79] GenomeInfoDb_1.38.1         withr_3.0.2                \n [81] fastmap_1.2.0               fansi_1.0.6                \n [83] SparseM_1.84-2              digest_0.6.37              \n [85] R6_2.6.1                    mime_0.13                  \n [87] textshaping_0.4.0           colorspace_2.1-1           \n [89] scattermore_1.2             tensor_1.5.1               \n [91] dichromat_2.0-0.1           spatstat.data_3.1-8        \n [93] generics_0.1.4              data.table_1.17.8          \n [95] graphlayouts_1.2.2          httr_1.4.7                 \n [97] htmlwidgets_1.6.4           S4Arrays_1.2.0             \n [99] uwot_0.2.3                  pkgconfig_2.0.3            \n[101] gtable_0.3.6                lmtest_0.9-40              \n[103] SingleCellExperiment_1.24.0 XVector_0.42.0             \n[105] htmltools_0.5.8.1           dotCall64_1.2              \n[107] scales_1.4.0                Biobase_2.62.0             \n[109] png_0.1-8                   spatstat.univar_3.1-4      \n[111] ggdendro_0.2.0              rjson_0.2.23               \n[113] reshape2_1.4.4              nlme_3.1-168               \n[115] zoo_1.8-14                  cachem_1.1.0               \n[117] GlobalOptions_0.1.2         stringr_1.5.2              \n[119] KernSmooth_2.23-26          parallel_4.3.3             \n[121] miniUI_0.1.2                logger_0.4.0               \n[123] pillar_1.11.0               grid_4.3.3                 \n[125] vctrs_0.6.5                 RANN_2.6.2                 \n[127] VGAM_1.1-13                 promises_1.3.3             \n[129] stringfish_0.17.0           xtable_1.8-4               \n[131] cluster_2.1.8.1             truncdist_1.0-2            \n[133] cli_3.6.5                   compiler_4.3.3             \n[135] crayon_1.5.3                future.apply_1.20.0        \n[137] labeling_0.4.3              plyr_1.8.9                 \n[139] stringi_1.8.7               viridisLite_0.4.2          \n[141] deldir_2.0-4                assertthat_0.2.1           \n[143] gsl_2.1-8                   lazyeval_0.2.2             \n[145] spatstat.geom_3.5-0         quantreg_6.1               \n[147] Matrix_1.6-5                RcppHNSW_0.6.0             \n[149] patchwork_1.3.2             future_1.67.0              \n[151] shiny_1.11.1                plotthis_0.10.0            \n[153] SummarizedExperiment_1.32.0 evd_2.3-7.1                \n[155] ROCR_1.0-11                 gridtext_0.1.5             \n[157] igraph_2.1.4                memoise_2.0.1              \n[159] RcppParallel_5.1.9          bracer_1.2.1               \n",time:"# Generated by pipen-runinfo v1.1.2\n\nCommand: stdbuf -oL Rscript /mnt/disks/.cwd/.pipen/Immunopipe/SeuratClusterStats/0/job.script\nVoluntary context switches: 891\nInvoluntary context switches: 258\nPercentage of CPU this job got: 123%\nMajor page faults: 0\nMinor page faults: 378146\nMaximum resident set size (kB): 1455824\nElapsed real time (s): 15.73\nSystem (kernel) time (s): 0.65\nUser time (s): 18.70\nExit status: 0\n",device:"# Generated by pipen-runinfo v1.1.2\n\nScheduler\n---------\nlocal\n\nHostname\n--------\n4807a2eabf37\n"}},{name:"ClusterMarkers",desc:"Markers for clusters of all or selected T/B cells.",npages:1,report_toc:!0,order:9},{name:"ClonalStats",desc:"Visualize the clonal information.",npages:1,report_toc:!0,order:10}]};function On(e){let t,n;return t=new Pt({props:{dir:"up"}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function zn(e){let t,n;return t=new Pt({props:{dir:"down"}}),{c(){h(t.$$.fragment)},m(e,r){$(t,e,r),n=!0},i(e){n||(l(t.$$.fragment,e),n=!0)},o(e){i(t.$$.fragment,e),n=!1},d(e){f(t,e)}}}function jn(e){let t,n,r,s,o,d=e[1]>0&&On();n=new Vn({});let u=e[1]<e[3].npages-1&&zn();return{c(){d&&d.c(),t=b(),h(n.$$.fragment),r=b(),u&&u.c(),s=g()},m(e,a){d&&d.m(e,a),_(e,t,a),$(n,e,a),_(e,r,a),u&&u.m(e,a),_(e,s,a),o=!0},p(e,n){e[1]>0?d?2&n&&l(d,1):(d=On(),d.c(),l(d,1),d.m(t.parentNode,t)):d&&(c(),i(d,1,1,()=>{d=null}),p()),e[1]<e[3].npages-1?u?10&n&&l(u,1):(u=zn(),u.c(),l(u,1),u.m(s.parentNode,s)):u&&(c(),i(u,1,1,()=>{u=null}),p())},i(e){o||(l(d),l(n.$$.fragment,e),l(u),o=!0)},o(e){i(d),i(n.$$.fragment,e),i(u),o=!1},d(e){e&&(a(t),a(r),a(s)),d&&d.d(e),f(n,e),u&&u.d(e)}}}function Fn(e){let t,n,r;function s(t){e[4](t)}let a={logo:e[0],logotext:e[3].desc,versions:Dn.versions,entries:Dn.entries,pipeline_name:Dn.pipeline.name,report_toc:e[3].report_toc,runinfo:e[3].runinfo,toc:Gn,page:e[1],$$slots:{default:[jn]},$$scope:{ctx:e}};return void 0!==e[2]&&(a.isSideNavOpen=e[2]),t=new Tt({props:a}),x.push(()=>v(t,"isSideNavOpen",s)),{c(){h(t.$$.fragment)},m(e,n){$(t,e,n),r=!0},p(e,[r]){const s={};1&r&&(s.logo=e[0]),8&r&&(s.logotext=e[3].desc),8&r&&(s.report_toc=e[3].report_toc),8&r&&(s.runinfo=e[3].runinfo),2&r&&(s.page=e[1]),42&r&&(s.$$scope={dirty:r,ctx:e}),!n&&4&r&&(n=!0,s.isSideNavOpen=e[2],y(()=>n=!1)),t.$set(s)},i(e){r||(l(t.$$.fragment,e),r=!0)},o(e){i(t.$$.fragment,e),r=!1},d(e){f(t,e)}}}function Hn(e,t,n){let r,{name:s}=t,{page:a=0}=t,o=!1;for(const e of Dn.entries)if(e.procs){for(const t of e.procs)if(t.name==s){r=t;break}}else if(e.name==s){r=e;break}return m(()=>{const e=new URL(window.location.href);let t;e.hash.length>1&&(t=document.getElementById(`${e.hash.slice(1)}`)),t&&t.scrollIntoView({block:"start",behavior:"smooth"})}),e.$$set=e=>{"name"in e&&n(0,s=e.name),"page"in e&&n(1,a=e.page)},[s,a,o,r,function(e){o=e,n(2,o)}]}e("default",new class extends t{constructor(e){super(),n(this,e,Hn,Fn,r,{name:0,page:1})}}({target:document.body,props:{name:"SeuratClusterStats",page:0}}))}}});
